Clean-up Bugzilla login dialog

jdatapple opened the pull request:
#### e2921a98a281f827b763564e581858f5ede0100c
<pre>
Clean-up Bugzilla login dialog
<a href="https://bugs.webkit.org/show_bug.cgi?id=283599">https://bugs.webkit.org/show_bug.cgi?id=283599</a>

Reviewed by Alexey Proskuryakov.

Sets the &quot;Remember my login&quot; to baseline alignment and
changes the login dialog to a modal, autofocusing the email
input field with refined sizing to accommodate more varied
mobile viewports.

Canonical link: <a href="https://commits.webkit.org/287223@main">https://commits.webkit.org/287223@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32726eebcf966349700302ad3ee80b055540514e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77861 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56894 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31236 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82509 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29118 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66054 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5189 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60968 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18915 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80929 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50954 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66870 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41270 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48328 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27461 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69421 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/24689 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83871 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5228 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3521 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69195 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/5384 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66779 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68447 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12448 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10569 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12169 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5176 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5168 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8600 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6953 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
rs=me

nt1m commented on Websites/bugs.webkit.org/template/en/custom/account/auth/login-small.html.tmpl:
```diff
@@ -37,7 +37,7 @@
         id="mini_login[% qs_suffix FILTER css_class_quote %]" class="mini_login dialog-grid">
         
-    <input id="Bugzilla_login[% qs_suffix FILTER css_class_quote %]" required
+    <input id="Bugzilla_login[% qs_suffix FILTER css_class_quote %]" required autofocus
```
`autofocus` here is redundant. The dialog focusing steps for modal dialogs should automatically pick the first focusable element and focus it.

jdatapple replied:
Right. However, the first focusable element that WebKit chooses is the "close" button, hence the need for the `autofocus` attribute on the login field.

jdatapple committed:
Clean-up Bugzilla login dialog
https://bugs.webkit.org/show_bug.cgi?id=283599

Reviewed by Alexey Proskuryakov.

Sets the "Remember my login" to baseline alignment and
changes the login dialog to a modal, autofocusing the email
input field with refined sizing to accommodate more varied
mobile viewports.

Canonical link: https://commits.webkit.org/287223@main
